clang/runtime/dpct-rt/include/dpct/graph.hpp

```diff
@@ -22,5 +22,56 @@ typedef sycl::ext::oneapi::experimental::command_graph<
     sycl::ext::oneapi::experimental::graph_state::executable>
     *command_graph_exec_ptr;
 
+namespace detail {
+class graph_mgr {
+public:
+  graph_mgr() = default;
+  graph_mgr(const graph_mgr &) = delete;
+  graph_mgr &operator=(const graph_mgr &) = delete;
+  graph_mgr(graph_mgr &&) = delete;
+  graph_mgr &operator=(graph_mgr &&) = delete;
+
+  static graph_mgr &instance() {
+    static graph_mgr instance;
+    return instance;
+  }
+
+  bool begin_recording(sycl::queue *queue_ptr) {
+    auto graph = new sycl::ext::oneapi::experimental::command_graph<
+        sycl::ext::oneapi::experimental::graph_state::modifiable>(
+        queue_ptr->get_context(), queue_ptr->get_device());
+    auto result = queue_graph_map.insert({queue_ptr, graph});
+    if (!result.second) {
+      return false;
+    }
+    return graph->begin_recording(*queue_ptr);
```

Comment:
Is the entry for the `queue_ptr` not updated as expected?
Also `graph` is not released in the code.

Reply:
Done.